packages/data-service/src/instance-detail-helper.ts

```diff
@@ -654,62 +748,69 @@ function getInstanceDetail(client, db, done) {
   async.auto(tasks, function (err, results) {
     if (err) {
       // report error
+      // @ts-expect-error Callback without result...
       return done(err);
     }
     // cleanup
-    results = omit(results, [
+    const cleanedResult = omit(results, [
       'db',
       'listDatabases',
       'allowedDatabases',
       'userInfo',
       'listCollections',
       'allowedCollections',
       'cmdLineOpts',
-    ]);
-    return done(null, results);
+    ]) as unknown as InstanceDetails;
+    return done(null, cleanedResult);
   });
 }
 
-function getInstance(client, db, done) {
+function getInstance(
+  client: MongoClient,
+  db: Db,
+  done: Callback<InstanceDetails>
+): void {
   getInstanceDetail(client, db, function (err, res) {
     if (err) {
+      // @ts-expect-error Callback without result...
       return done(err);
     }
 
-    let port;
-    let hostname;
-
-    if (has(db, 's.options.url')) {
-      debug(
-        'parsing port and hostname from driver url option `%s`',
-        db.s.options.url
-      );
-      port = URL.port(db.s.options.url);
-      hostname = URL.hostname(db.s.options.url);
-    }
-
-    if (has(res, 'host.hostname')) {
-      /**
-       * Use the hostname from getBuildInfo() as authoritative.
-       *
-       * NOTE (imlucas) In the real world, `db.s.options.url`
-       * gives a guarantee that the hostname is routable
-       * from the users perspective.  The hostname returned by
-       * getBuildInfo() may not (e.g. AWS' internal DNS for EC2).
-       */
-      hostname = URL.hostname(res.host.hostname);
-      if (/:\d+/.test(res.host.hostname)) {
-        port = URL.port(res.host.hostname);
-      }
-    }
-
-    res._id = [hostname, port].join(':');
+    // let port = '';
+    // let hostname = '';
+
+    // if (has(db, 's.options.url')) {
+    //   debug(
+    //     'parsing port and hostname from driver url option `%s`',
+    //     (db as any).s.options.url
+    //   );
+    //   port = URL.port((db as any).s.options.url);
+    //   hostname = URL.hostname((db as any).s.options.url);
+    // }
+
+    // if (has(res, 'host.hostname')) {
+    //   /**
+    //    * Use the hostname from getBuildInfo() as authoritative.
+    //    *
+    //    * NOTE (imlucas) In the real world, `db.s.options.url`
+    //    * gives a guarantee that the hostname is routable
+    //    * from the users perspective.  The hostname returned by
+    //    * getBuildInfo() may not (e.g. AWS' internal DNS for EC2).
+    //    */
+    //   hostname = URL.hostname(res.host.hostname);
+    //   if (/:\d+/.test(res.host.hostname || '')) {
+    //     port = URL.port(res.host.hostname);
+    //   }
+    // }
```

Comment:
Is this just not needed anymore (because of the new driver version or something)?

Reply:
Ah yes, forgot to remove it. The thing is that this is only need to set the `_id` field - and it gets set by the callback of the single call that is made to the method directly so it's redundant!

Reply:
Got it, thanks for the context!